### PR TITLE
Feature/file upload field and confirm mode

### PR DESCRIPTION
**Pull request by jpbu**
* Introduce new field type for file upload
* add handling for confirm before submit
* add documentation and example for form fields to README

### CODE REVIEW

**Review by wisam-alhennawi — APPROVED** [review]
Thank you, I tested it, it looks good.

I also fixed a bug in the `select` component, could you please check it?

File: src/LaraFormComponent.php [hunk 1]
```diff
@@ -3,10 +3,11 @@
 namespace WisamAlhennawi\LaraFormsBuilder;
 
 use Livewire\Component;
+use Livewire\WithFileUploads;
 
 abstract class LaraFormComponent extends Component
 {
-    use LaraFormsBuilder;
+    use LaraFormsBuilder, WithFileUploads;
```
[the remark this answers is not in the document]
Comment (jpbu) [reply src/LaraFormComponent.php:10]: I think if we provide/include a form field with type file that only works with this trait we should make sure it already works out of the box without additional steps. Do you have concerns regarding a performance penalty of unused traits or why do you think it shouldn't be always set? If that is problematic, I would suggest to provide an extended separate `LaraFormsWithFileUploadComponent` or similar as a more convenient way.

**Comment by jpbu** [comment]
> I also fixed a bug in the `select` component, could you please check it?

Thanks, I double checked and validated the fix, looks good and fixes the error of multi-word label display in view/confirm mode.

File: src/LaraFormComponent.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (wisam-alhennawi) [reply src/LaraFormComponent.php:10]: No, it is not problematic, it is just regarding unused traits. but we can merge it.

**Pull request merged**